tests/test_core_network.py

```diff
@@ -2335,6 +2335,40 @@ def test_subnetwork_on_spatial_condition_delagates_to_spatial_methods_to_get_sub
     )
 
 
+def test_removing_mode_from_all_links_retains_other_modes_on_links():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike", "piggyback"}, "length": 1})
+    n.add_link("1", 2, 3, attribs={"modes": {"car", "bike"}, "length": 1})
+
+    n.remove_mode_from_all_links(mode="bike")
+
+    assert n.link("0")["modes"] == {"car", "piggyback"}, "The link modes have been incorrectly set"
+    assert n.link("1")["modes"] == {"car"}, "The link modes have been incorrectly set"
+
+
+def test_removing_mode_from_all_links_deletes_links_with_no_other_modes():
+    n = Network("epsg:27700")
+    n.add_link("0", 1, 2, attribs={"modes": {"car", "bike"}, "length": 1})
+    n.add_link("1", 2, 3, attribs={"modes": {"bike"}, "length": 1})
```

Comment:
This is not directly about this PR, but this test raises a question: Is there some warning in the logs or elsewhere when you remove all modes from a link and that link is subsequently removed from the graph? I get that a link with no modes cannot be traversed, and so can be removed, but the user might have done that inadvertently, and the downstream effects could be very confusing if so.

Reply:
There are logs when links are removed. The removal of links is not automatic, unless you use these dedicated methods. So a user could mess around with modes on the links and the structure of the graph remains the same. In either case, whether the links are removed or not (so, just modes missing from links) genet has methods to verify validity of the graph and if they pose issues these will come out in the reports. These run automatically if using genet cli commands, or can/should be called upon if a user is using genet as a package and expecting to use their network with matsim.